src/nvim/os/env.c

```diff
@@ -34,7 +34,30 @@ const char *os_getenv(const char *name)
 
 int os_setenv(const char *name, const char *value, int overwrite)
 {
+#ifdef HAVE_SETENV
   return setenv(name, value, overwrite);
+#elif defined(HAVE_PUTENV_S)
+  if (!overwrite && os_getenv(name) != NULL) {
+    return 0;
+  }
+  if (_putenv_s(name, value) == 0) {
+    return 0;
+  }
+  return -1;
+#else
+  // Use putenv, instead of setenv - this will leak memory
+  // but in some systems only putenv is available
+  if (!overwrite && os_getenv(name) != NULL) {
+    return 0;
+  }
+
+  size_t len = strlen(name) + 1 + strlen(value) + 1;
+  char *env = xmalloc(len);
+  xstrlcpy(env, name, len);
+  STRCAT(env, "=");
+  STRCAT(env, value);
```

Comment:
Can't just regular `strcat` be used? I don't know anything about character encoding on Windows.